Star Seeker/Views/Compositions/MovementController/JoystickMovementController.swift

```diff
@@ -105,22 +130,123 @@ class JoystickMovementController : MovementController {
+        
+        let path = CGMutablePath()
+        path.addLines(between: points)
+        
+        let shapeNode = SKShapeNode(path: dashedPath)
+        shapeNode.strokeColor = .white
+        shapeNode.lineWidth = 3.0
+        shapeNode.zPosition = 20
+        
+        return shapeNode
+        
+    }
+    
+    static func createDashedPath(points: [CGPoint], lineWidth: CGFloat, dashLength: CGFloat, gapLength: CGFloat) -> CGPath {
+        let path = CGMutablePath()
+        if (points.count == 0) {
+            return path
+        }
+        var currentPoint = points[0]
+        path.move(to: currentPoint)
+
+        var isDrawing = true
+        var distanceRemaining: CGFloat = dashLength
+        
+        for i in 0..<points.count {
+            let nextPoint = points[i]
+            let segmentVector = CGVector(dx: nextPoint.x - currentPoint.x, dy: nextPoint.y - currentPoint.y)
+            let segmentLength = sqrt(segmentVector.dx * segmentVector.dx + segmentVector.dy * segmentVector.dy)
+            var segmentProgress: CGFloat = 0
+
+            while segmentProgress < segmentLength {
+                let drawLength = min(distanceRemaining, segmentLength - segmentProgress)
+                let drawVector = CGVector(dx: segmentVector.dx * drawLength / segmentLength, dy: segmentVector.dy * drawLength / segmentLength)
+                let drawPoint = CGPoint(x: currentPoint.x + drawVector.dx, y: currentPoint.y + drawVector.dy)
+
+                if isDrawing {
+                    path.addLine(to: drawPoint)
+                } else {
+                    path.move(to: drawPoint)
+                }
+
+                currentPoint = drawPoint
+                segmentProgress += drawLength
+                distanceRemaining -= drawLength
+
+                if distanceRemaining <= 0 {
+                    isDrawing = !isDrawing
+                    distanceRemaining = isDrawing ? dashLength : gapLength
+                }
+            }
+        }
+
+        return path
+    }
+    
+    static func calculateTrajectoryPoints(impulse: CGVector, startingPosition: CGPoint) -> [CGPoint] {
+        var points = [CGPoint]()
+        
+        let gravity = GameConfig.playerGravity
+        let mass = GameConfig.playerMass
+        let initialVelocity = CGVector(dx: impulse.dx * mass, dy: impulse.dy * mass)
+        
+        let numberOfPoints = 50
+        let timeStep = 0.1
+        
+        for i in 0..<numberOfPoints {
+            let t = Double(i) * timeStep
+            let x = initialVelocity.dx * CGFloat(t)
+            let y = initialVelocity.dy * CGFloat(t) + 0.5 * gravity.dy * CGFloat(t * t)
+            let point = CGPoint(x: x, y: y)
+            if (point.getDistance() > startingPosition.getDistance()) {
+                points.append(point)
+            }
+        }
+        
+        return points
+    }
+    
+    static func arrowControllerNode (buttonSize: Double) -> SKSpriteNode {
```

Comment:
arrowControllerNode basically means the joystick knob

Reply:
the joystick know pull guidance for the user be able to pull down intuitively from visual, to be precise